packages/aws-cdk-lib/aws-lambda-nodejs/lib/bundling.ts

```diff
@@ -133,8 +133,15 @@ export class Bundling implements cdk.BundlingOptions {
     // Don't automatically externalize aws sdk if `bundleAwsSDK` is true so it can be
     // include in the bundle asset
     const defaultExternals = props.runtime?.isVariable || props.bundleAwsSDK ? [] : versionedExternals;
+
     const externals = props.externalModules ?? defaultExternals;
 
+    // warn users if they are using a runtime that does not support sdk v2
+    // and the sdk is not explicitly bundled
+    if (externals.length && isV2Runtime) {
```

Comment:
what is `externals.length` supposed to mean/indicate?

Reply:
`externals` are the packages we are explicitly not bundling. `externals.length` should check if that list has anything in it. `externals` will have the sdk package correlated to the runtime by default. Users can override or provide their own external packages.

The thought process behind this is that if a user is setting `bundleAwsSDK = true` or `props.externalModules = []`, then they likely are aware of how the bundling in the module works, and they are including the sdk in their handler code. Both of these will cause `externals.length` to be falsy, and not give the warning since they are avoiding sdk problems by bundling it in.

If they have not set anything, or set certain externals, we give them this warning.